src/ensembl/src/content/app/browser/track-panel/track-panel-modal/modal-views/TrackPanelBookmarks.test.tsx

```diff
@@ -96,41 +113,32 @@ describe('<TrackPanelBookmarks />', () => {
   });
 
   it('shows the ellipsis only when the total objects is more than 20', () => {
-    const previouslyViewedObjects = times(20, () =>
+    const newPreviouslyViewedObjects = times(20, () =>
       createPreviouslyViewedLink()
     );
-    const { rerender } = render(
-      <TrackPanelBookmarks
-        {...props}
-        previouslyViewedObjects={previouslyViewedObjects}
-      />
-    );
+
+    (useSelector as any).mockReturnValue(newPreviouslyViewedObjects);
+
+    const { rerender } = render(<TrackPanelBookmarks />);
     const previouslyViewedSection = screen.getByText('Previously viewed');
 
     expect(previouslyViewedSection.querySelector('button')).toBeFalsy();
 
     // Add another link to make it 21 links
-    previouslyViewedObjects.push(createPreviouslyViewedLink());
-    rerender(
-      <TrackPanelBookmarks
-        {...props}
-        previouslyViewedObjects={previouslyViewedObjects}
-      />
-    );
+    newPreviouslyViewedObjects.push(createPreviouslyViewedLink());
+    rerender(<TrackPanelBookmarks />);
 
     expect(previouslyViewedSection.querySelector('button')).toBeTruthy();
   });
 
   it('calls changeDrawerViewAndOpen when the ellipsis is clicked', () => {
-    const previouslyViewedObjects = times(21, () =>
+    const newPreviouslyViewedObjects = times(21, () =>
```

Comment:
I would much rather reuse the same variable 😞 I find `newPreviouslyViewedObjects` more confusing than reusing the variable from the outer scope. @msmanoj might think differently.

Reply:
I've used mockState so this is now redundant ✅